Added new settings to disable certain GUI menu items.

BittyByte opened the pull request:
Hi, I modified the program and added the following *new* settings into the default.properties file:

HIDE_LOGIN_AS_INVISIBLE = false
DISABLE_PREFERENCES_MENU_ITEM = false
MAINT_FILESPEC =
DISABLE_PRESENCE_STATUS_BAR = false
DISABLE_OPEN_HIDE_STATUS_IN_SYSTRAY = false
DISABLE_BROADCAST_MENU_ITEM = false
DISABLE_AVATAR_TAB = false
DISABLE_PLUGINS_MENU_ITEM = false

The MAINT_FILESPEC feature allows an administrator to override the DISABLE_PREFERENCES_MENU_ITEM setting. The "Preferences" menu item can be re-enabled when the specified file exists. It's probably best to place this file on a read/only network share.

Please consider merging this into the master so that others can benefit from these much asked-for program features.

Thanks,

Michael

wrooot commented:
You've only added two out of eight new settings to the Default.java file.

speedy01 commented:
perhaps some of this could be integrated with client control plugin for openfire?  I believe that plugin may already disable things like broadcast?

wrooot commented:
Oh, that would be terrific to have all these options in the Admin Console and being able to control them without modifying Spark. Wonder who would take on such a huge task :)

wrooot commented:
@BittyByte as i have said in the forums, i will have to test that myself (properties working without adding them to default.java). As for the Open\Hide menus. I would like to leave them as is and only hide the Presence menu in the System tray.

wrooot commented:
Oh, and making the presence box without a dropdown and only showing the presence bubble and presence text would be great, if you can achieve that.

BittyByte commented:
Hi Speedy01,
If added to the "Client Control" plugin, I assume that would introduce the ability to allow/disallow the setting for _all_ users not just specific users? Or would can administrator specify the setting per user?

wrooot commented:
Currently Client Control plugin enables\disables features for all users. Having groups based settings would be even greater, but even greater task :)

speedy01 commented:
I think in its current state, the plug in is all or nothing. Since openfire keeps everything in a database, I'm sure there could be a way make user specific changes if someone knew what they were doing... I am not that person!

BittyByte commented:
Hi Wroot,
Yes, I agree it would be ideal if _all_ of the Spark client settings can be controlled from a central location such as an admin console plugin to openfire. But as a network admin, this was something I had to do at request of management in a short amount of time, which is why I simply modified the Spark source.

BittyByte commented:
Hi Wroot,
I will make a new pull request today to leave the Open/Hide options as is in the system tray icon. I agree it shouldn't be controlled by a setting. However, logically it doesn't really make sense to why it was included in the source code to begin with.

wrooot commented:
Michael, me and speedy are only dreaming aloud. There is nothing wrong with your approach ;)  as for the hide/open menus, well i think it was added to have a visual counterpart to the clicking shortcut. Sometimes we get requests in the forums to make Spark window go up by double clicking the icon on the desktop :)  there are various degrees of users in the wild.

wrooot commented:
Btw, by making a new PR you meant updating this one? It would be better. If you can't do that, then closing this one and opening new one with a complete list of changes would be better.

BittyByte commented:
Hi Wroot,
I was going to do a "commit and push" from Eclipse to my fork. What do I do next if not a pull request? Remember, I'm still a noob at this.

BittyByte commented:
Hi Wroot,
I made the changes for Open/Hide as you requested. Should I also add the other 6 settings to the Default.java file? It's strange that some settings from default.properties are not included in that file, yet still work ok. For example, DISABLE_EXIT is also missing, yet work just fine. While other settings are required to be in there. Anyway, I'll include them if you want before I do my next "commit and push". Just let me know after you're done testing. Thanks.

wrooot commented:
I'm also only a sysadmin :)  i think if you do a pull request it will create new one based on this one and as this one is not merged yet, it will create a conflict later. I'm not verse with git cmd, so i can only think of closing this PR and then doing new one with the recent changes included. But do not do this yet. I will probably have some time tomorrow to test this.

BittyByte commented:
OK, in the meantime I will try to figure out how to properly disable the presence status bar so it is not clickable. Let me know tomorrow after your testing how to upload all my changes the right way. Thanks again for all your help.

BittyByte commented:
Hi Wroot,

Good news!! I made all the modifications as you requested including properly disabling the presence status bar as well leaving the Open/Hide alone while still being able to disable the Status option in the system tray. Here is a summary of changes so far:

- Changed **DISABLE_OPEN_HIDE_STATUS_IN_SYSTRAY** to **DISABLE_STATUS_IN_SYSTRAY**

- Modified option to properly disable the presence status bar by making it _unclickable_, removing the _bevel frame_, as well as removing the _down arrow_.

Let me know if you want the 6 remaining settings added to the Default.java file and when you're done with testing. And of course, how to upload the changes I made.

Regards,

Michael

wrooot commented:
What is your username in JIRA? I just want to assign the ticket to you, so i won't take the credit and if someone wants to comment, they will know who the author of this patch is :)  There is some weird glitch with my internet provider and i can't access Github from my pc. So i have a limited access from my tablet. But i should be able to review and merge the PR.

BittyByte commented:
I created a new PR #233. Can you provide me with the correct link to JIRA? I'm not sure if I signed up to the correct site. The one I used is: [https://id.atlassian.com/](url) and used my email address as userid

akrherz commented:
@BittyByte our issue tracker is https://issues.igniterealtime.org

BittyByte commented:
OK, just signed up to the correct site. My username is BittyByte.

BittyByte commented:
Hi Wroot,
Here is why you may be having trouble accessing GitHub (Massive DNS outage):

http://gizmodo.com/this-is-probably-why-half-the-internet-shut-down-today-1788062835

BittyByte commented:
OK so it seems Travis CI is currently having "upstream connectivity issues impacting builds":

https://www.traviscistatus.com/incidents/m41c9md5dg9z

So eventually my PR should make it through the tests and merge successfully. I want to thank everyone, especially Wroot, for helping me with my first PR. Have a nice weekend guys!

Regards,

Michael

wrooot commented:
Oh i see. Explains a lot. I have even stopped to receive emails from here so i didn't even know the new PR is available and didn't see you question about Jira.

BittyByte commented:
You may be able to regain access if you change your DNS settings to:

**8.8.8.8 and 208.67.222.222**

BittyByte commented:
Hi Wroot,
Regarding the issues you brought up, I added my comments here:
https://issues.igniterealtime.org/browse/SPARK-1822

BittyByte commented:
Hi Wroot,

I submitted a new PR #234 with fixes for the issues you brought to my attention. Thanks again for all your help!
